sophie/DE_analysis.R

```diff
@@ -0,0 +1,205 @@
+  # * `DESeqDataSet()`: For RangedSummarizedExperiment (RSE) data, like from recount2
+  #
+  # Documentation recommends using `tximport` and then calling `DESeqDataSetFromTximport` for RSEM (or other methods listed above)
+  # estimated count data in order to get counts corrected for by gene length: https://support.bioconductor.org/p/94003/#94028.
+  #
+  # [tximport](https://rdrr.io/bioc/tximport/src/R/tximport.R):
+  # * imports transcript-level estimates from various external software and optionally summarizes abundances,
+  # counts, and transcript lengths to the gene-level (default) or transcript-level matrices.
+  #
+  # _Input_ a list of files containing:
+  # * transcript-level abundances (TPM/FPKM): Normalized read count. Number of reads mapped to a given
+  # transcript normalized by transcript length and/or library size.
+  # * estimated/expected counts: An estimate of the number of reads derived from a given transcript/isoform
+  # or gene. We can only get an estimate for this quantity since reads do not often map uniquely to a single transcript/isoform.
+  # * effective lengths: The number of possible start positions for a read within a transcript
+  #
+  # _Output_ a list of matrices containing:
+  # If the user requests just the transcript-level data, then combine data across samples and return.
+  # Otherwise summarizes to gene-level using `tx2gene` parameter: a two-column dataframe linking transcript id (column 1) to gene id (column 2).
+  # * abundance matrix
+  # * count matrix
+  # * length matrix
+  #
+  # For the analysis in notebook: explore_uncorrelated_genes/1a_DE_polyA_vs_ribo.ipynb
+  # We are using data generated by RSEM, which returned gene-level summaries.
+  # In this case, `tximport` will just combine the gene-level summaries across files into a matrix.
+  #
+  # The dataset we have is a matrix of estimated counts that were combined across samples. We don't have the individual raw data files output from RSEM,
+  # which is needed to call tximport. However, it looks like `tximport` would have been redundant given what we mentioned above.
+  # Here we are rounding the combined RSEM estimated counts and using `DESeqDataSetFromMatrix()`.
+  # Given the first step of `DESeqDataSetFromTximport` is rounding the counts, we anticipate this step to be ok.
+  # https://rdrr.io/bioc/DESeq2/src/R/AllClasses.R.
+  #
+  # In general, the recommendation to use `tximport` might be necessary for the case where we have transcript-level data.
+  #
+  # For the analysis in notebook: human_general_analysis/2_identify_generic_genes_pathways.ipynb
+  # We are using RangedSummarizedExperiments from recount2.
+  # We combined these RSEs in download_recount2_data.R so we cannot use `DESeqDataSet()` which expects SE inputs.
+  # Again, we round our combined RSE data and use `DESeqDataSetFromMatrix()`.
+  # Looks like rounding is also performed for `DESeqDataSet()` as well.
+
```

Comment:
You need to add this line, or else the differential expression results will be wrong

```suggestion
metadata$group <- as.factor(metadata$group)
```